modules/api/src/test/java/org/apache/fluo/api/data/BytesTest.java

```diff
@@ -79,6 +79,19 @@ public void testHashSubsequence() {
   }
 
   @Test
+  public void testPrefixSuffix() {
+    Bytes b1 = Bytes.of("abcde");
+    Bytes prefix = Bytes.of("ab");
+    Bytes suffix = Bytes.of("de");
+    Bytes mid = Bytes.of("cd");
+
+    Assert.assertEquals(true, b1.startsWith(prefix));
+    Assert.assertEquals(true, b1.endsWith(suffix));
+    Assert.assertEquals(false, b1.startsWith(mid));
+    Assert.assertEquals(false, b1.endsWith(mid));
```

Comment:
Some test cases that compare with self would be nice.  I have had problems in the past with implementations that do not properly handle this case.  Also, test with subsequence (which will create a new Bytes that has the same inner array) would be nice.

```java
  assertTrue(b1.startsWith(b1));
  assertTrue(b1.endsWith(b1));
  assertTrue(b1.startsWith(b1.subSequence(0,2));
  assertFalse(b1.subSequence(0,2).startsWith(b1));
  //do similar test with subseq and endswith
```